wled00/fcn_declare.h

```diff
@@ -165,8 +173,10 @@ bool parseLx(int lxValue, byte* rgbw);
 void parseLxJson(int lxValue, byte segId, bool secondary);
 
 //mqtt.cpp
+#ifndef WLED_DISABLE_MQTT
```

Comment:
For all changes you propose in fcn_declare.h, the same is true: removing prototype declarations saves exactly zero bytes.
Even with prototypes, there will be linker errors when the function implementation is missing. So I see no good reason to remove these prototypes.

Reply:
Removing the prototypes by itself indeed does very little, but it gives compile errors that can then be removed by marking the non-functional code with the appropriate wrappings. 

As a second step I went into the declaration sites and ensured they and their headers were also removed.

Overall those #ifndef / #ifdef locations could be made more readable if they were properly abstracted away.
But that is not the state of the project and went beyond what I needed to achieve.

All I know is that at the end of this PR the build was small enough to allow an OTA update using a workaround I found in one of the other PRs. While still allowing for websockets so that Home Assistant works.